odoo/fields.py

```diff
@@ -2572,7 +2572,10 @@ def get_values(self, env):
         if isinstance(selection, str):
             selection = getattr(env[self.model_name], selection)()
         elif callable(selection):
-            selection = selection(env[self.model_name])
+            model = env[self.model_name]
+            if 'lang' in env.context:
+                model = model.with_context(lang=False)
```

Comment:
```suggestion
            model = env[self.model_name].with_context(lang=None)
```

Reply:
Shouldn't we avoid environment creations as much as possible ? (i.e. not update the context when not necessary)

Reply:
It really depends on how often this is called.
The performance gain is probably ridiculous, and in that case is not worth the code complexity.
But I am not sure about the performance gain, so I invite you to benchmark the alternatives to see :wink:

Reply:
Integrated some query counts tests, with the difference before & after the change :wink: